internal/controller/standalone_pgadmin/pgadmin_controller.go

```diff
@@ -0,0 +1,227 @@
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+// PGAdminReconciler reconciles a PGAdmin object
+type PGAdminReconciler struct {
+	client.Client
+	Owner  client.FieldOwner
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=postgres-operator.crunchydata.com,resources=pgadmins,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=postgres-operator.crunchydata.com,resources=pgadmins/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=postgres-operator.crunchydata.com,resources=pgadmins/finalizers,verbs=update
+
+// Reconcile which aims to move the current state of the cluster closer to the
+// desired state described in a [v1beta1.PGUpgrade] identified by request.
+func (r *PGAdminReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	log := logging.FromContext(ctx)
+
+	pgAdmin := &v1beta1.PGAdmin{}
+	if err := r.Get(ctx, req.NamespacedName, pgAdmin); err != nil {
+		if err = client.IgnoreNotFound(err); err != nil {
+			log.Error(err, "unable to fetch PGAdmin")
+		}
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, r.reconcilePGAdminStatefulSet(ctx, pgAdmin)
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PGAdminReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1beta1.PGAdmin{}).
+		Complete(r)
+}
+
+// reconcilePGAdminStatefulSet writes the StatefulSet that runs pgAdmin.
+func (r *PGAdminReconciler) reconcilePGAdminStatefulSet(
+	ctx context.Context, pgadminCR *v1beta1.PGAdmin,
+) error {
+	sts := &appsv1.StatefulSet{ObjectMeta: standalonePGAdmin(pgadminCR)}
+	sts.SetGroupVersionKind(appsv1.SchemeGroupVersion.WithKind("StatefulSet"))
+
+	sts.Annotations = naming.Merge(
+		pgadminCR.Spec.Metadata.GetAnnotationsOrNil(),
+		pgadminCR.Spec.Metadata.GetAnnotationsOrNil())
+	sts.Labels = naming.Merge(
+		pgadminCR.Spec.Metadata.GetLabelsOrNil(),
+		pgadminCR.Spec.Metadata.GetLabelsOrNil(),
+		commonLabels(pgadminCR),
+	)
+	sts.Spec.Selector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			LabelStandalonePGAdmin: pgadminCR.Name,
+			LabelRole:              roleStandalonePGAdmin,
+		},
+	}
+	sts.Spec.Template.Annotations = naming.Merge(
+		pgadminCR.Spec.Metadata.GetAnnotationsOrNil(),
+		pgadminCR.Spec.Metadata.GetAnnotationsOrNil())
+	sts.Spec.Template.Labels = naming.Merge(
+		pgadminCR.Spec.Metadata.GetLabelsOrNil(),
+		pgadminCR.Spec.Metadata.GetLabelsOrNil(),
+		commonLabels(pgadminCR),
+	)
+
+	// Don't clutter the namespace with extra ControllerRevisions.
+	sts.Spec.RevisionHistoryLimit = initialize.Int32(0)
+
+	// Give the Pod a stable DNS record based on its name.
+	// - https://docs.k8s.io/concepts/workloads/controllers/statefulset/#stable-network-id
+	// - https://docs.k8s.io/concepts/services-networking/dns-pod-service/#pods
+	sts.Spec.ServiceName = standalonePGAdminService(pgadminCR).Name
```

Comment:
Just flagging this: we don't reconcile a service yet.

Reply:
OBE re: refactor